### ISSUE TITLE
[BUG/ISSUE] Cannot run GEOS-Chem Classic continuously across year boundary

### ISSUE THREAD

**Issue by lizziel** [geoschem/geos-chem#305]
When running GEOS-Chem Classic across a year boundary I get the following run-time error:

```
HISTORY (INIT): Opening ./HISTORY.rc
=====================================================================
GEOS-Chem ERROR: No diagnostic output will be created for collection: 
"Restart"!  Make sure that the length of the simulation as specified in 
"input.geos" (check the start and end times) is not shorter than the frequency 
setting in HISTORY.rc!  For example, if the frequency is 010000 (1 hour) but 
the simulation is set up to run for only 20 minutes, then this error will occur.
 -> at History_ReadCollectionData (in module History/history_mod.F90)

 -> ERROR occurred at (or near) line     70 of the HISTORY.rc file
=====================================================================

=====================================================================
GEOS-Chem ERROR: Error encountered in "History_ReadCollectionData"!
 -> at History_Init (in module History/history_mod.F90)
=====================================================================

=====================================================================
GEOS-CHEM ERROR: Error encountered in "History_Init"!
STOP at  -> at GEOS-Chem (in GeosCore/main.F90)
=====================================================================
```

For this particular run my settings in input.geos are as follows:

```
Start YYYYMMDD, hhmmss  : 20151231 230000
End   YYYYMMDD, hhmmss  : 20160101 000000
```

I tested various combinations of start and end dates and consistently have this error ONLY if the start and end have a change in year. The quick fix is to edit HISTORY.rc to change all instances of frequency and duration set as `'End' `to be a specific time duration, such as `010000` (hourly). 

We will look into a longer term solution to put into a future version.

**Comment by yantosca** [comment]
Thanks for letting us know about this issue.  I see what is happening.  I set up a 2-day transport tracer simulation (12/31/15 to 1/2/16) that straddles the new year.  The simulation length = 2 days = 176400 sec.  But the file write alarm is 3153600 seconds, because this is computed from the start of the current year (2015).  

I think the fix is to test if you straddle a year, and then subtract off the number of seconds in that year.  That will reduce the filewritealarm.  Apparently this is an edge case that I didn't foresee.  I'll work on a fix now.  Stay tuned.


The error happens for the Restart collection, which has frequency "end"

**Comment by yantosca** [comment]
This issue is now resolved by commit https://github.com/geoschem/geos-chem/commit/2fcf78e56e877828054ec7622cc9ac2364369eb5.  We have added an extra check to make sure that the FileWriteAlarm for containers with freq=End are computed properly.

**Comment by neyranaz** [comment]
In the context of [this](https://github.com/geoschem/geos-chem-cloud/issues/40) project, I incorporated to GC12.7.0 the correction suggested in this thread to CodeDir/History/history_util_mod.F90 and I was successful at transitioning from year 2014 to 2015 when generating boundary conditions in an Aerosol Only Run. `geosfp_4x5_aerosol`

In contrast, when performing the actual nested grid run in a `geosfp_025x3125_aerosol_na` directory, I used the same correction but I was unsuccessful in transitioning from 2014 to 2015 and obtained the same run-time error as the one shown at the beginning of this thread.  Perhaps this correction might not work on a Nested Grid Run.  

To discard any other error, I ran a nested run for a same-year period (2014), using the same specifications as when I tried to transition from 2014 to 2015 and I was successful in doing so. So there might be an intrinsic issue with running a nested simulation across a year boundary. Are there any ideas on how this could be solved?
